### ISSUE TITLE
[Feature]: Support Async Module Dist Path Customization

### ISSUE THREAD

**Issue by joealden** [web-infra-dev/rsbuild#1699]
### What problem does this feature solve?

Currently (from what I can tell), it isn't possible to remove or change the `async/` directory prefix for async module chunks (those created via dynamic imports) without using [`tools.rspack`](https://rsbuild.dev/config/tools/rspack) (via [`output.chunkFilename`](https://www.rspack.dev/config/output.html#outputchunkfilename)).

From my testing, async module placement respects the `js` property of the [`output.distPath`](https://rsbuild.dev/config/output/dist-path) config option, but there is no way to place them at the same level as the regular JS module chunks (or any other location). As mentioned above, I can work around this, but I'm thinking it'd be nice if this was configurable in `rsbuild` directly?

Note that in my project, all async chunks use `webpackChunkName` magic comments to name them explicitly, so the risk of potential naming conflicts is low. Also, the existing `output.distPath.server` and `output.distPath.worker` properties could suffer from naming collisions if a user configured `rsbuild` in a way where these were at the same level. I mention this in case you might be concerned about what allowing this could mean.

Relevant doc links:

- https://rsbuild.dev/config/output/filename#filename-of-async-modules
- https://rsbuild.dev/guide/optimization/split-chunk#using-dynamic-import-for-code-splitting

### What does the proposed API look like?

I think there are multiple ways to approach this, but as `output.distPath` already has the concept of `root` (and how other properties are relative to this), maybe the `js` option type could be `string | { root?: string; async?: string }`? The downside of this would be that it'd be a little cumbersome to have async chunks in a totally different directory to other JS files (as users would have to use `../`), but for this kind of edge case that doesn't seem like a big API problem.

Alternatively, if we want to keep `output.distPath` flat, an `asyncJs`/`jsAsync` property could be added?

**Comment by chenjiahan** [comment]
I totally agree with this proposal. In fact, we also wanted to add a path option for async js chunks.

I prefer the flat style, we can add the `jsAsync` and `cssAsync` options:

- type:

```ts
type DistPathConfig = {
  root?: string;
  js?: string;
  css?: string;
}
```

**Comment by chenjiahan** [comment]
I totally agree with this proposal. In fact, we also wanted to add a path option for async js chunks.

I prefer the flat style, we can add the `jsAsync` and `cssAsync` options:

- Type:

```ts
type DistPathConfig = {
  js?: string;
  jsAsync?: string;

  css?: string;
  cssAsync?: string;

  // ...other paths
}
```

- Default:

```js
const defaultDistPath = {
  js: 'static/js',
  jsAsync: `${distPath.js}/async`

  css: 'static/css',
  cssAsync: ${distPath.css}/async`

  // ...other paths
};
```

I will implement this option soon.

**Comment by joealden** [comment]
Thank you - amazing how fast your team adds things!

**Comment by chenjiahan** [comment]
Released in v0.4.9 😄
